File: Labs/Lab2/Lab2.py
import re

def isAnagram(s, t):
    if (not s or not t):
        print("Input is null")
        return False
    s = re.sub("[^a-z]", "", s.lower())
    t = re.sub("[^a-z]", "", t.lower())
    letters = {}
    for c in s:
        if not letters.get(c):
            letters.update({c : 0})
        letters.update({c : letters.get(c)+1})
    for c in t:
        if not letters.get(c):
            return False
        letters.update({c : letters.get(c)-1})
        if letters.get(c) < 0:
            return False
    for i in letters.values():
        if i != 0:
            return False
    return True

print(isAnagram("Anagram", "Nag a ram"))
print(isAnagram("ABCDE", "DADBC"))
print(isAnagram(None, "null"))

File: Labs/Lab2/Lab2.java
package Labs.Lab2;

public class Lab2 {
    public static boolean isAnagram(String s, String t) {
        if (s == null || t == null) {
            System.err.println("Input is null");
            return false;
        }
        s = s.toLowerCase().replaceAll("[^a-z]", "");
        t = t.toLowerCase().replaceAll("[^a-z]", "");
        int[] letters = new int[26];
        for (int i = 0; i < s.length(); i++) {
            int current = s.charAt(i)-'a';
            letters[current]++;
        }
        for (int i = 0; i < t.length(); i++) {
            int current = t.charAt(i)-'a';
            letters[current]--;
            if (letters[current] < 0) {
                return false;
            }
        }
        for (int i : letters) {
            if (i != 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(isAnagram("Anagram", "Nag a ram"));
        System.out.println(isAnagram("ABCDE", "DADBC"));
        System.out.println(isAnagram(null, "null"));
    }
}
